components/brave_ads/core/internal/ads/serving/eligible_ads/pipelines/inline_content_ads/eligible_inline_content_ads_v3.cc

```diff
@@ -0,0 +1,144 @@
+#include "brave/components/brave_ads/common/interfaces/ads.mojom-shared.h"
+#include "brave/components/brave_ads/core/internal/ads/ad_events/ad_events_database_table.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/choose/predict_ad.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/eligible_ads/eligible_ads_features.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/eligible_ads/exclusion_rules/exclusion_rules_util.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/eligible_ads/exclusion_rules/inline_content_ads/inline_content_ad_exclusion_rules.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/targeting/user_model_info.h"
+#include "brave/components/brave_ads/core/internal/ads_client_helper.h"
+#include "brave/components/brave_ads/core/internal/common/logging_util.h"
+#include "brave/components/brave_ads/core/internal/creatives/inline_content_ads/creative_inline_content_ads_database_table.h"
+#include "brave/components/brave_ads/core/internal/geographic/subdivision/subdivision_targeting.h"
+#include "brave/components/brave_ads/core/internal/resources/behavioral/anti_targeting/anti_targeting_resource.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace brave_ads::inline_content_ads {
+
+EligibleAdsV3::EligibleAdsV3(
+    geographic::SubdivisionTargeting* subdivision_targeting,
+    resource::AntiTargeting* anti_targeting_resource)
+    : EligibleAdsBase(subdivision_targeting, anti_targeting_resource) {}
+
+void EligibleAdsV3::GetForUserModel(
+    targeting::UserModelInfo user_model,
+    const std::string& dimensions,
+    GetEligibleAdsCallback<CreativeInlineContentAdList> callback) {
+  BLOG(1, "Get eligible inline content ads");
+
+  database::table::AdEvents database_table;
+  database_table.GetForType(
+      mojom::AdType::kInlineContentAd,
+      base::BindOnce(&EligibleAdsV3::OnGetForUserModel, base::Unretained(this),
+                     std::move(user_model), dimensions, std::move(callback)));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void EligibleAdsV3::OnGetForUserModel(
+    targeting::UserModelInfo user_model,
+    const std::string& dimensions,
+    GetEligibleAdsCallback<CreativeInlineContentAdList> callback,
+    const bool success,
+    const AdEventList& ad_events) {
+  if (!success) {
+    BLOG(1, "Failed to get ad events");
+    std::move(callback).Run(/*had_opportunity*/ false, {});
+    return;
+  }
+
+  GetBrowsingHistory(std::move(user_model), ad_events, dimensions,
+                     std::move(callback));
+}
+
+void EligibleAdsV3::GetBrowsingHistory(
+    targeting::UserModelInfo user_model,
+    const AdEventList& ad_events,
+    const std::string& dimensions,
+    GetEligibleAdsCallback<CreativeInlineContentAdList> callback) {
+  const int max_count = features::GetBrowsingHistoryMaxCount();
+  const int days_ago = features::GetBrowsingHistoryDaysAgo();
```

Comment:
See other versions as these have been removed and moved to line 71, check all instances